### PR TITLE
Test for Voucher without any codes

### PR DESCRIPTION
**Pull request by michal-macioszczyk**
Add test for voucher without any codes, potential edge case to check if Voucher.used resolver works properly.

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

File: saleor/graphql/discount/tests/mutations/test_voucher_create.py [hunk 1]
```diff
@@ -118,6 +118,55 @@ def test_create_voucher(
...
+            "type": VoucherTypeEnum.ENTIRE_ORDER.name,
+            "addCodes": [""],
+            "discountValueType": DiscountValueTypeEnum.FIXED.name,
+            "minCheckoutItemsQuantity": 10,
+            "startDate": start_date.isoformat(),
+            "endDate": end_date.isoformat(),
+            "applyOncePerOrder": True,
+            "applyOncePerCustomer": True,
+            "singleUse": True,
+            "usageLimit": 3,
```
Comment (zedzior) [inline saleor/graphql/discount/tests/mutations/test_voucher_create.py:146]: I see conflict here: `singleUse=True` while `usageLimit=3`. Are we sure it doesn't impact calculations? I.e `saleor.discount.utils.increase_voucher_usage` function will perform both `increase_voucher_code_usage_value()` and `deactivate_voucher_code()`.

Comment (michal-macioszczyk) [reply saleor/graphql/discount/tests/mutations/test_voucher_create.py:146]: this combination will work but in this case we do not touch `increase_voucher_usage` because we just want to create Voucher and check if resolver on Voucher field `used` will work properly.

**Review by zedzior — APPROVED** [review]

File: saleor/graphql/discount/tests/mutations/test_voucher_create.py [hunk 2]
```diff
@@ -118,6 +118,55 @@ def test_create_voucher(
...
+    staff_api_client,
+    permission_manage_discounts,
+    product,
+    variant,
+    collection,
+    category,
+):
+    # given
+    start_date = timezone.now() - timedelta(days=365)
+    end_date = timezone.now() + timedelta(days=365)
```
Comment (Air-t) [inline saleor/graphql/discount/tests/mutations/test_voucher_create.py:131]: in general shouldn't we use `freeze_time` when working with datetimes?

**Review by Air-t — APPROVED** [review]
